src/Components/Endpoints/src/Discovery/PageComponentBuilder.cs

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// Initializes a new instance of <see cref="PageComponentBuilder"/>.
+/// </summary>
+[DebuggerDisplay($"{{{nameof(GetDebuggerDisplay)}(),nq}}")]
+public class PageComponentBuilder : IEquatable<PageComponentBuilder?>
+{
+    private IReadOnlyList<string> _routeTemplates = Array.Empty<string>();
+
+    /// <summary>
+    /// Gets or sets the source for this page component.
+    /// </summary>
+    public required string Source { get; set; }
```

Comment:
Same for the other properties.

Also, should this type name really include `Builder`? The API doesn't appear to follow a builder-like pattern.

Reply:
I'd rather not make my life harder with init. Since these APIs are consumed by the source generator and are not something that regular users are going to consume.

Reply:
Totally fine - definitely no need for `init` if it doesn't fit the scenario conveniently.

May still be good to rename to remove `Builder` though since it's not like a builder. Maybe `PageComponentMetadata`.